identity/webapp/src/utility/encode-registration-formData.ts

```diff
@@ -0,0 +1,54 @@
+import { sign, verify } from 'jsonwebtoken';
+
+// we need some jwt encoding to deal with passing data to an auth0 action
+// https://auth0.com/docs/customize/actions/flows-and-triggers/login-flow/redirect-with-actions#pass-data-back-to-auth0
+
+type TokenPayload = {
+  iat: number;
+  iss: string;
+  sub: string;
+  exp: number;
+  aud?: string;
+  state?: string;
+  other?: FormDataObject;
+};
```

Comment:
You get this - and other type-checking - for free with `@types/jsonwebtoken`; eg [here](https://github.com/wellcomecollection/identity/blob/ccb4b02df0391553ed876fd6353c7ed38e334ba8/packages/apps/api-authorizer/src/authorization.ts#L1)

Reply:
@jamieparkinson  I don't think JWT as a standard mandates any of the members, whereas specific implementations do, so e.g. the `JwtPayload` interface looks like this

```
export interface JwtPayload {
    [key: string]: any;
    iss?: string | undefined;
    sub?: string | undefined;
    aud?: string | string[] | undefined;
    exp?: number | undefined;
    nbf?: number | undefined;
    iat?: number | undefined;
    jti?: string | undefined;
}
```
but Auth0 requires that `iat`, `iss`, `sub` and `exp` are present (and feels like it would make sense to make `other` in our implementation too).

So does it make sense to override the types in `@types/jsonwebtoken`, or write our own like this?